File: pset6/readability/readability.py
from cs50 import get_string

text = get_string("Input text you wish to grade: ")

letters = 0
words = 1
sentences = 0

for i in range(len(text)):
    if text[i] >= 'A' and text[i] <= 'Z' or text[i] >= 'a' and text[i] <= 'z':
        letters += 1
    elif text[i] == ' ':
        words += 1
    elif text[i] == '?' or text[i] == '!' or text[i] == '.':
        sentences += 1

L = (letters / words) * 100
S = (sentences / words) * 100

grade = round(0.0588 * L - 0.296 * S - 15.8)

if grade >= 16:
    print("Grade 16+")
elif grade < 1:
    print("Before Grade 1")
else:
    print("Grade", grade)

File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>


int main(void)

{
   //Prompt for a sentence.
   string sentence  = get_string("Text: ");

   int letters = 0;
   int words = 1;
   int sentences = 0;

   for (int i = 0, n = strlen(sentence); i < n; i++)

   //Calculate amount of letters.
   if (isalpha(sentence[i]))
       {
       letters++;
       }

    //Amount of words.
    else if (isspace(sentence[i]))
      {
       words++;
      }

    //Amount of sentences.
    else if (sentence[i] == '.' || sentence[i] == '!' || sentence[i] == '?')
       {
       sentences++;
       }

    //Calculate the grade level.
    float L = letters / (float)words * 100;
    float S = sentences / (float)words * 100;
    int index = round((0.0588 * L) - (0.296 * S) - 15.8);

    //Print Grade Level
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}



